apps/condo/migrations/20220510155000-0119_manual_set_resident_in_ticket_client_field.js

```diff
@@ -0,0 +1,40 @@
+exports.up = async (knex) => {
+    await knex.raw(`
+        BEGIN;
+            UPDATE "Ticket"
+            SET "client" = "Resident"."user",
+                "clientName" = "User"."name",
+                "clientPhone" = "User"."phone",
+                "clientEmail" = "User"."email"
+            FROM "Contact"
+            JOIN "User" ON
+                "Contact"."phone" = "User"."phone"
+            JOIN "Resident" ON
+                "Resident"."user" = "User"."id"
+            WHERE "Ticket"."client" IS NULL AND
+                  "Ticket"."contact" = "Contact"."id" AND
+                  "Resident"."property" = "Ticket"."property" AND
+                  "Resident"."unitName" = "Ticket"."unitName" AND
+                  "Resident"."unitType" = "Ticket"."unitType";
+        COMMIT;
+    `)
+}
+
+exports.down = async (knex) => {
+    await knex.raw(`
+        BEGIN;
+            UPDATE "Ticket"
+            SET "client" = NULL
+            FROM "Contact"
+            JOIN "User" ON
+                "Contact"."phone" = "User"."phone"
+            JOIN "Resident" ON
+                "Resident"."user" = "User"."id"
+            WHERE "Ticket"."client" IS NULL AND
```

Comment:
you're trying to set "Ticket"."client" to NULL but it would already be NULL, what is the meaning?

Reply:
there should be IS NOT NULL, fixed